### ISSUE TITLE
Add option for mapping between int64 and JS number

### ISSUE THREAD

**Issue by Mike-Bell** [emscripten-core/emscripten#20950]
It would be nice if there were a compiler flag for automatically downcasting / upcasting a JS `number` to a wasm int64. JS numbers have a precision between 32 and 64 bits (54 bits I believe). If I know my application needs more than 32 bits but less than 54 bits, it would be nice to just work with numbers rather than having to have bigints all over the place. This would have to be an opt-in option since it would cause overflow in cases where numbers exceed JS's max size.

**Comment by sbc100** [comment]
I agree it would be nice to have this option.   For wasm64 we do this automatically, at least for JS library functions, when we know the arguments are pointers or size it.   We create wrappers in that case for our JS library functions. See https://github.com/emscripten-core/emscripten/blob/bc5998833dcd0f48e90a8cb13fdf40e36480e4cb/src/jsifier.js#L123.

For exported nextive function we currently have a horrible hardcoded list here: https://github.com/emscripten-core/emscripten/blob/bc5998833dcd0f48e90a8cb13fdf40e36480e4cb/tools/emscripten.py#L873-L884

I'd love to find a way to avoid this list.

In the long run I'd really love JS numbers to automatically convert to i64 on the boundary.   We already do this for i32 (which silently truncates floating pointers).    There is discussion on that here: https://github.com/WebAssembly/JS-BigInt-integration/issues/12
